src/bindings/gaiabuild/view.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+import { Env } from '../../lib/env';
+import { qps } from '../../lib/pseudo';
+import { getResourceLinks } from '../../bindings/html/head';
+import { translateFragment } from '../../bindings/html/dom';
+import { getDirection } from '../../bindings/html/langs';
+import { serializeContext } from '../../bindings/gaiabuild/serialize';
+
+export class View {
+  constructor(htmloptimizer, fetch) {
+    this.env = new Env(
+      htmloptimizer.config.GAIA_DEFAULT_LOCALE, fetch);
+    this.htmloptimizer = htmloptimizer;
+    this.doc = htmloptimizer.document;
+    this.ctx = this.env.createContext(getResourceLinks(this.doc.head));
+
+    // add the url of the currently processed webapp to all errors
+    this.env.addEventListener('*', amendError.bind(this));
+
+    this.stopBuildError = null;
+    const log = logError.bind(this);
+    const stop = stopBuild.bind(this);
+
+    // stop the build if these errors happen for en-US
+    // XXX tv_apps break the build https://bugzil.la/1179833
+    // this.env.addEventListener('fetcherror', stop);
+    // XXX parse errors don't have the lang property needed by stopBuild
```

Comment:
Yeah, I agree with you although I'm not sure how exactly the intercepting should look like.  I'm fine leaving this to be figured out at a later date;  we don't do this in l10n.js right now, so we don't need to introduce this new error report right now.